tests/MqttClientTest.cpp

```diff
@@ -223,242 +155,483 @@ static int s_TestIoTMqtt311ConnectWithNoSigningCustomAuth(Aws::Crt::Allocator *a
+ */
+static int s_TestMqtt311DirectConnectionWithTLS(Aws::Crt::Allocator *allocator, void *)
+{
+    struct aws_string *endpoint = NULL;
+    struct aws_string *port = NULL;
+
+    int error = aws_get_environment_value(allocator, s_mqtt311_test_envName_direct_tls_hostname, &endpoint);
+    error |= aws_get_environment_value(allocator, s_mqtt311_test_envName_direct_tls_port, &port);
+
+    bool isEveryEnvVarSet = (endpoint && port);
+    if (isEveryEnvVarSet == true)
     {
-        printf("Failed to create MQTT311 client from config");
-        ASSERT_TRUE(false);
+        isEveryEnvVarSet = (aws_string_is_valid(endpoint) && aws_string_is_valid(port));
     }
-    auto connection = client.NewConnection(clientConfig);
-    if (!*connection)
+    if (error != AWS_OP_SUCCESS || isEveryEnvVarSet == false)
     {
-        printf("Failed to create MQTT311 connection from config");
-        ASSERT_TRUE(false);
+        printf("Environment Variables are not set for the test, skip the test");
+        aws_string_destroy(endpoint);
+        aws_string_destroy(port);
+        return AWS_OP_SKIP;
     }
 
-    std::promise<bool> connectionCompletedPromise;
-    std::promise<void> connectionClosedPromise;
-    auto onConnectionCompleted =
-        [&](Aws::Crt::Mqtt::MqttConnection &, int errorCode, Aws::Crt::Mqtt::ReturnCode returnCode, bool) {
-            (void)returnCode;
-            if (errorCode)
-            {
-                connectionCompletedPromise.set_value(false);
-            }
-            else
-            {
-                connectionCompletedPromise.set_value(true);
-            }
-        };
-    auto onDisconnect = [&](Aws::Crt::Mqtt::MqttConnection &) { connectionClosedPromise.set_value(); };
-    connection->OnConnectionCompleted = std::move(onConnectionCompleted);
-    connection->OnDisconnect = std::move(onDisconnect);
-
-    Aws::Crt::UUID Uuid;
-    Aws::Crt::String uuidStr = Uuid.ToString();
+    Aws::Crt::ApiHandle apiHandle(allocator);
 
-    if (!connection->Connect(uuidStr.c_str(), true /*cleanSession*/, 5000 /*keepAliveTimeSecs*/))
+    Aws::Crt::Io::TlsContextOptions tlsCtxOptions = Aws::Crt::Io::TlsContextOptions::InitDefaultClient();
+    ASSERT_TRUE(tlsCtxOptions);
+    tlsCtxOptions.SetVerifyPeer(false);
+    Aws::Crt::Io::TlsContext tlsContext(tlsCtxOptions, Aws::Crt::Io::TlsMode::CLIENT, allocator);
+    ASSERT_TRUE(tlsContext);
+
+    Aws::Crt::Mqtt::MqttClient client;
+    Aws::Crt::Io::SocketOptions socketOptions;
+    socketOptions.SetConnectTimeoutMs(3000);
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> connection = client.NewConnection(
+        aws_string_c_str(endpoint), (uint16_t)std::stoi(aws_string_c_str(port)), socketOptions, tlsContext, false);
+    int connectResult = s_ConnectAndDisconnect(connection);
+    if (connectResult != AWS_OP_SUCCESS)
     {
-        printf("Failed to connect");
-        ASSERT_TRUE(false);
+        throw std::runtime_error("ConnectAndDisconnect failed");
```

Comment:
Is there any reason that we choose to use `runtime_error` instead of the original pattern with `ASSERT`?

Reply:
It's because, for some reason I wasn't able to reproduce or quite determine, `ASSERT_TRUE` would be counted as a skip in Codebuild rather than an error. I tried to reproduce it in aws-c-common but was unable to, and then I forgot to look into it further as I moved on to Javascript at the time.

I'm not sure what was causing it though. We can probably go back to ASSERT though - I fixed the issues causing the tests to fail.